packages/cozy-konnector-libs/libs/linkBankOperations.js

```diff
@@ -170,9 +173,83 @@ class Linker {
       })
     })
     .then(() => {
-      return result
+      const notLinkedBills = this.getNotLinkedBills(result)
+      const billsGroups = this.groupBillsByOriginalDate(notLinkedBills)
+
+      const combinations = flatten(
+        Object
+          .values(billsGroups)
+          .map(billsGroup => this.generateBillsCombinations(billsGroup))
+      )
+
+      const combinedBills = combinations.map(
+        combination => this.combineBills(...combination)
+      )
+
+      return Promise.all(combinedBills.map(bill => {
+        return findDebitOperation(this.cozyClient, bill, options, allOperations)
```

Comment:
findDebitOperation is synchronous now ? Maybe we could get rid of all this Promise handling ?